gammacat/collection.py

```diff
@@ -256,27 +262,42 @@ def process_datasets(self):
                         dataset.write(path)
 
     def make_index_file_for_input(self):
-        # TODO: this is a temp hack. Fill correctly using GammaCatResourceIndex
-        data = OrderedDict()
-        data['info'] = gammacat_info.info_dict
-        data['data'] = self.input_data.datasets.to_dict()['data']
-        data['files'] = self.config.list_of_files()
-        path = self.config.index_datasets_json
-        write_json(data, path)
+        resources = []
+        for info_filename in self.input_data.info_yaml_list:
+            info_data = load_yaml(info_filename)
+            if info_data['data_entry']['status'] == 'missing':
+                continue
+            # TODO: Decide which datasets are copied to output collection by the keywords in 
+            # 'status' and 'reviewed' in info.yaml
+            # e.g if info_data['data_entry']['status'] == 'complete':
+            for dataset in info_data['datasets']:
+                if dataset.endswith('yaml'):
+                    resource = DataSet.read(info_filename.parent / dataset).resource
+                elif dataset.endswith('ecsv'):
+                    try:
```

Comment:
This try except is no good, because `Lightcurve.read` might work on other ECSV files now, or in the future.
One way to do it would be to look for the "lc" or "sed" at the appropriate place in the filename string.

Reply:
Here's an example that shows that it currently doesn't work -- an SED is put as type: lightcurve in the index:
https://github.com/pdeiml/gamma-cat/blob/c91b69cf39d71811dfcc380082623791e74629f3/input/input-datasets.json#L13

Reply:
Ok, but I don't know how to distinguish between a sed- and a light curve file.
I tried
elif dataset.endswith('lc*.ecsv')
....
elif dataset.endswith('sed*.ecsv')
...
but that does not work. I need this placeholder before the .ecsv because there can be multiple sed files in a folder.

Reply:
You should write a helper function, something like this:
```
def resource_type_from_filename(filename):
    return filename.split('.')[0].split('-')[-1]
```
To write such code, it's best to do it interactively in ipython
```
$ ipython
In [1]: filename = 'tev-000138-sed.ecsv'
```
OK, in this case it seems to be a one-liner, and if you only need it in one place in the code, you can just put the statement there. but if it gets more complex (e.g. because some filenames you pass there don't work with that expression and need to be filtered out) or if you use it multiple times, then it would be the time to split it out into a helper function.

Reply:
Will not work.
For a file: tev-000034-sed-1.ecsv
filename.split('.')[0].split('-')[-1]
will lead to    1
while for
tev-000034-sed.ecsv
it will lead to   sed

So you would need something like
if ..endswith('.ecsv')
if filename.split('.')[0].split('-')[-1] == 'sed':
...
else:
if filename.split('.')[0].split('-')[-2] == 'sed':
... 
if filename.split('.')[0].split('-')[-2] == 'lc':
...

This will become extremely messy!

The problem is the "moving" position of the keywords 'lc' and 'sed' within the filename string!